{{cpp/title|bad_variant_access}}
{{cpp/utility/variant/navbar}}
{{ddcl|since=c++17|header=variant|
class bad_variant_access : public std::exception
}}

{{tt|std::bad_variant_access}} 是下列情形中抛出的异常的类型：
* 以不匹配当前活跃可选项的索引或类型调用 {{ltt|cpp/utility/variant/get|std::get{{small|(std::variant)}}}}。
* 调用 {{rlpt|visit2|std::visit}}探访一个 {{rlpt|valueless_by_exception}} 的变体。
{{rrev|since=c++26|
* 调用 {{rlpt|visit|std::variant::visit}} 探访一个 {{rlpt|valueless_by_exception}}的 {{tt|variant}}。
}}

===成员函数===
{{cpp/error/exception/member of derived|bad_variant_access|std=c++17}}

{{cpp/error/exception/inherit}}

===示例===
{{example
|code=
#include &lt;iostream&gt;
#include &lt;variant&gt;

int main()
{
    std::variant&lt;int, float&gt; v;
    v = 12;
    try
    {
        std::get&lt;float&gt;(v);
    }
    catch(const std::bad_variant_access&amp; e)
    {
        std::cout &lt;&lt; e.what() &lt;&lt; '\n';
    }
}
|p=true
|output=
bad_variant_access
}}

===参阅===
{{dsc begin}}
{{dsc inc|cpp/utility/variant/dsc get}}
{{dsc inc|cpp/utility/variant/dsc visit2}}
{{dsc inc|cpp/utility/variant/dsc visit}}
{{dsc end}}

{{langlinks|en|es|ja|ru}}